flatsurf/geometry/thurston_veech.py

```diff
@@ -0,0 +1,157 @@
+r"""
+Thurston-Veech construction
+
+A Thurston-Veech surface is a translation surface whose horizontal and vertical
+directions are parabolic, that is stabilized by a multitwist. Combinatorially,
+such surface decomposes into rectangles with horizontal and vertical sides.
+
+We encode such surface by labelling the rectangles by `\{1, 2, \ldots, n\}`.
+Then, the gluings is described by two permutations `h` and `v` that describe
+respectively horizontal gluings (going right) and vertical gluings (going up).
+The horizontal and vertical cylinders are then respectively the cycles
+of `h` and `v`.
+
+As shown by Thurston and Veech, to fix the flat structure of the surface one
+only has to specify the topological data of the twists, that is their
+multiplicities in each cylinder.
+
+REFERENCES:
+
+- Pascal Hubert, Erwan Lanneau
+  "Veech groups without parabolic elements"
+"""
+#*****************************************************************************
```

Comment:
You probably want to use the same copyright header here.

Reply:
yep. Copied from the wrong place.